sycl/include/sycl/ext/oneapi/experimental/complex/detail/marray.hpp

```diff
@@ -0,0 +1,316 @@
+    }                                                                          \
+    return lhs;                                                                \
+  }                                                                            \
+                                                                               \
+  friend marray &operator op(marray & lhs, const value_type & rhs) {           \
+    for (std::size_t i = 0; i < NumElements; ++i) {                            \
+      lhs[i] op rhs;                                                           \
+    }                                                                          \
+    return lhs;                                                                \
+  }                                                                            \
+  friend marray &operator op(value_type & lhs, const marray & rhs) {           \
+    for (std::size_t i = 0; i < NumElements; ++i) {                            \
+      lhs[i] op rhs;                                                           \
+    }                                                                          \
+    return lhs;                                                                \
+  }
+
+  MARRAY_CPLX_OP(+=)
+  MARRAY_CPLX_OP(-=)
+  MARRAY_CPLX_OP(*=)
+  MARRAY_CPLX_OP(/=)
+
+#undef MARRAY_CPLX_OP
+
+  // MARRAY_CPLX_OP is: %=
+  friend marray &operator%=(marray &lhs, const marray &rhs) = delete;
+  friend marray &operator%=(marray &lhs, const value_type &rhs) = delete;
+  friend marray &operator%=(value_type &lhs, const marray &rhs) = delete;
+
+// MARRAY_CPLX_OP is: ++, --
+#define MARRAY_CPLX_OP(op)                                                     \
+  friend marray operator op(marray &lhs, int) = delete;                        \
+  friend marray &operator op(marray & rhs) = delete;
+
+  MARRAY_CPLX_OP(++)
+  MARRAY_CPLX_OP(--)
+
+#undef MARRAY_CPLX_OP
+
+// MARRAY_CPLX_OP is: unary +, unary -
+#define MARRAY_CPLX_OP(op)                                                     \
+  friend marray<value_type, NumElements> operator op(                          \
+      const marray<value_type, NumElements> &rhs) {                            \
+    marray<value_type, NumElements> rtn;                                       \
+    for (std::size_t i = 0; i < NumElements; ++i) {                            \
+      rtn[i] = op rhs[i];                                                      \
+    }                                                                          \
+    return rtn;                                                                \
+  }
+
+  MARRAY_CPLX_OP(+)
+  MARRAY_CPLX_OP(-)
+
+#undef MARRAY_CPLX_OP
+
+// MARRAY_CPLX_OP is: &, |, ^
+#define MARRAY_CPLX_OP(op)                                                     \
+  friend marray operator op(const marray &lhs, const marray &rhs) = delete;    \
+  friend marray operator op(const marray &lhs, const value_type &rhs) = delete;
```

Comment:
Is there a reason we don't delete
```c++
friend marray operator op(const value_type &lhs, const marray &rhs)
```
like we do for % and bitshifts? Seems like we could just have a `MARRAY_CPLX_OP_DELETE` and just delete all of them in one... Or two if we do the same for op-assigments.

Reply:
Sure! As I said in my previous comment. This was mostly arranged by type as a personal preference.
I'm happy to have a generic macro which deletes all operators.

Reply:
Please correct me if I did not understand correctly your feedback. Still, the reason for not having a generic `macro` is because of the different arguments that all operators received.

We can't place all the operators in the same basket due to their arguments.